vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/HelperIT.java

```diff
@@ -15,27 +15,86 @@
  */
 package com.vaadin.flow.component.checkbox.tests;
 
+import com.vaadin.flow.component.checkbox.testbench.CheckboxGroupElement;
+import com.vaadin.flow.component.checkbox.testbench.CheckboxElement;
 import com.vaadin.flow.testutil.TestPath;
 import com.vaadin.testbench.TestBenchElement;
 import com.vaadin.tests.AbstractComponentIT;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
+import java.util.List;
+
 @TestPath("vaadin-checkbox/helper")
 public class HelperIT extends AbstractComponentIT {
 
+    @Before
+    public void init() {
+        open();
+    }
+
     /**
      * Assert that helper component exists after setItems.
      * https://github.com/vaadin/vaadin-checkbox/issues/191
      */
     @Test
     public void assertHelperComponentExists() {
-        open();
         TestBenchElement checkboxGroup = $("vaadin-checkbox-group").first();
 
         TestBenchElement helperComponent = checkboxGroup.$("span")
                 .attributeContains("slot", "helper").first();
         Assert.assertEquals("Helper text", helperComponent.getText());
 
     }
+
+    @Test
+    public void assertCheckboxGroupHelperGenerator() {
+        CheckboxGroupElement group = $(CheckboxGroupElement.class)
+                .id("cbg-helper-generator");
+
+        List<CheckboxElement> checkboxes = group.getCheckboxes();
+        Assert.assertEquals("A helper", checkboxes.get(0).getHelperText());
+        Assert.assertEquals("B helper", checkboxes.get(1).getHelperText());
+        Assert.assertEquals("C helper", checkboxes.get(2).getHelperText());
+
+        $("button").id("clear-helper-generator").click();
+
+        Assert.assertEquals("", checkboxes.get(0).getHelperText());
+        Assert.assertEquals("", checkboxes.get(1).getHelperText());
+        Assert.assertEquals("", checkboxes.get(2).getHelperText());
+    }
+
+    @Test
+    public void assertCheckboxSlottedHelper() {
```

Comment:
```suggestion
    public void checkboxSlottedHelper() {
```

I'd probably disagree with adding a test for checking that a slotted helper is created. That's WC functionality, so checking the helper text property is enough IMO.

Reply:
Yes, initially it wasn't there but I added it to handle https://github.com/vaadin/flow-components/pull/6166#discussion_r1548043165.